src/cdomains/apron/linearTwoVarEqualityDomain.apron.ml

```diff
@@ -732,27 +732,26 @@ struct
       ) d;
       List.rev !acc *)
 
-  let invariant t =
-    match t.d with
-    | None -> []
-    | Some d ->
-      let acc = ref [] in
-      Array.iteri (fun i (var_opt, const) ->
-          let xi = Environment.var_of_dim t.env i in
-          let coeff_vars =
-            [(Coeff.s_of_int (-1), xi)] @ (match var_opt with
-                | None -> []
-                | Some var_index ->
-                  let var = Environment.var_of_dim t.env var_index in
-                  [(Coeff.s_of_int 1, var)])
-          in
-          let cst = Coeff.s_of_int (Z.to_int const) in
-          let lincons = Lincons1.make (Linexpr1.make t.env) Lincons1.EQ in
-          Lincons1.set_list lincons coeff_vars (Some cst);
-          acc := lincons :: !acc
-        ) d;
-      List.rev !acc
-
+      let invariant t =
+        match t.d with
+        | None -> []
+        | Some d ->
+            Array.fold_left (fun acc (i, (var_opt, const)) ->
+                let xi = Environment.var_of_dim t.env i in
+                let coeff_vars = 
+                    [(Coeff.s_of_int (-1), xi)] @ (match var_opt with
+                        | None -> []
+                        | Some var_index ->
+                            let var = Environment.var_of_dim t.env var_index in
```

Comment:
In case that we encounter var_index = i , we will not produce useful information. This should not end up as a list element. This is filtered out by the surrounding relational analysis framework, but nonetheless this creates extra load.